### DISCUSSION TITLE
Change ImageContent to BlobContent

### DISCUSSION THREAD

**Discussion by gabayben** [modelcontextprotocol/.github#89] [Ideas]
### Pre-submission Checklist

- [X] I have verified this would not be more appropriate as a feature request in a specific repository
- [X] I have searched existing discussions to avoid duplicates

### Your Idea

I feel like having just ImageContent is limiting, and could be a mistake that should be prevented early on.

This is especially redundant considering that it represents base64 encoded data, and the mime type is required anyway.

I think that it should be changed to BlobContent, like so:

`
export interface BlobContent extends Annotated {
  type: "blob";
  /**
   * The base64-encoded blob data.
   *
   * @format byte
   */
  data: string;
  /**
   * The MIME type of the blob. Different providers may support different mime types.
   */
  mimeType: string;
}
`

### Scope

- [X] Protocol Specification
- [ ] SDK Features
- [ ] Documentation
- [ ] Developer Experience
- [ ] Other

**Comment by jspahrsummers** [comment]
Thanks, this makes sense! Would you be willing to open a PR to fix this in the spec? We can incorporate it in the next draft version.

I'm going to close this out, as this would be better as an issue or PR, but very much agree with the point.

**Reply by gabayben** [reply]
Hey sorry for the late reply. Yes of course I will open a PR for this.
